src/exception/exception-api.ts

```diff
@@ -84,24 +74,23 @@ class ExceptionController {
   async clearProgramException(req: Request, res: Response) {
     const programId = req.params.programId;
     const result = await exceptionService.operations.deleteProgramException({ programId });
-    return res.status(getResStatus(result)).send(result);
+    return res.status(result.success ? 200 : 400).send(result);
   }
 
   @HasFullReadAccess()
   async getProgramException(req: Request, res: Response) {
     const programId = req.params.programId;
     const result = await exceptionService.operations.getProgramException({ programId });
-    return res.status(getResStatus(result)).send(result);
+    return res.status(result.success ? 200 : 400).send(result);
   }
 
+  // entity exceptions
   @HasFullWriteAccess()
   async createEntityException(req: Request, res: Response) {
     const programId = req.params.programId;
-    const existingProgramException = await exceptionService.operations.getProgramException({
-      programId,
-    });
+    const doc = await programExceptionRepository.find(programId);
 
-    if (existingProgramException.exception !== undefined) {
+    if (doc && doc.exceptions !== undefined) {
```

Comment:
would this achieve the same?
```suggestion
    if (doc?.exceptions !== undefined) {
```

Reply:
`    if (doc?.exceptions && doc.exceptions.length > 0) `
good call. yes the optional chaining works better. should be checking size of array too, not just if exists

Reply:
in that case `doc?.exceptions?.length > 0` would work too, right?

Reply:
`if (doc.exceptions && doc.exceptions.length > 0)` doing the same thing. no need for the optional chain on the doc?.exceptions

Reply:
you're right but TS wants same types :)